### ISSUE TITLE
mutex.cc error during tensorboard streaming

### ISSUE THREAD

**Issue by holgerroth** [NVIDIA/NVFlare#266]
When streaming training loss with 8 clients in secure mode in the [cifar10 example](https://github.com/NVIDIA/NVFlare/tree/main/examples/cifar10#5-streaming-tensorboard-metrics-to-the-server), a client can sometimes fail with one, both, or none of the below messages.

```
site-5/log.txt:[mutex.cc : 435] RAW: Lock blocking 0x7fbdb40b27b8   @
site-5/log.txt:[mutex.cc : 1908] RAW: Check (v & (kMuWait | kMuWrWait)) != kMuWrWait failed: Lock: Mutex corrupt: waiting writer with no waiters: 0x7fbdb40b1ff0
```

The below message will be written to the client log many times until the process is stopped manually.
```
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock returning 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Unlock 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock returning 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Unlock 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock returning 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Unlock 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock returning 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Unlock 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock returning 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Unlock 0x7fa63001b3d8   @
[mutex.cc : 435] RAW: Lock blocking 0x7fa63001b3d8   @
```
Mostly, the system becomes unstable after this error and needs to be shut down manually, e.g. using this [script](https://github.com/NVIDIA/NVFlare/blob/main/examples/cifar10/shutdown_fl_run.sh).

The issue sometimes only happens after training for many rounds (70-100 epochs in cifar10) and is more likely to happen in FL simulations within resource-restricted environments, e.g. a 1x 16 GB GPU instance on NGC. Also happens regularly a machine with Ubuntu 20.04.4 LTS (Intel® Core™ i7-7800X CPU @ 3.50GHz × 12).

It is also more likely to happen in secure mode than in poc mode.

Produced with nvflare==2.0.14

**Comment by YuanTingHsieh** [comment]
@holgerroth are we stilling seeing this? If so feel free to re-open, thanks!
